[DOC] Update Trademark Checklist to reflect Core vs Subsystem Trademarks (#630)
chipsalliance/Caliptra: doc/trademark/CaliptraChecklistAndEvaluationMethodology.md

Reviewer: We should add a requirement for OCP LOCK. Maybe something like:

```md
### *OCP LOCK Integration*                                                                                                                            

* **Checklist Item:** 
* **Requirement:** Any product integrating OCP LOCK MUST comply with all requirements listed in [lock_spec.ocp](../ocp_lock/lock_spec.ocp).
* **Evaluation Methodology:** Manufacturers evidence to demonstrate that the SoC complies with all OCP LOCK requirements.
```

Author: Will add

Author: To add a 3rd Caliptra Configuration that is Trademarkable (LOCK differentiated from Subsystem)

Author: See 8474e5468c73c06ed672e034c916f843dc9765ec

Author: @zhalvorsen please mark this resolved unless you have more edits to propose on it.

Reviewer: Looks great! Thanks Caleb!

- [Introduction](#introduction)
- [Caliptra Documentation](#caliptra-documentation)
- [Checklist Summary](#checklist-summary)
- [Part I: Common Requirements](#part-i-common-requirements)
  - [Assets](#assets)
  - [Interfaces](#interfaces)
  - [SoC Root of Trust (RoT)](#soc-root-of-trust-rot)
  - [Specific Functionality](#specific-functionality)
  - [Secure Processes](#secure-processes)
- [Part II: Caliptra Core Trademark — Additional Requirements (Passive Mode)](#part-ii-caliptra-core-trademark--additional-requirements-passive-mode)

> - [Part III: Caliptra Subsystem Trademark — Additional Requirements](#part-iii-caliptra-subsystem-trademark--additional-requirements)

Reviewer 2: Is reference link in document?  I don't see it in this PR.

Author: You're asking about the ToC link? It's valid.

  * **Requirement:** The IDEVID certificates, including CA certificates, MUST adhere to specified formats and cryptographic standards (e.g. key sizes and algorithms as per NIST guidelines). Acceptable algorithms and key strengths MUST meet or exceed the security strength that the Caliptra IP produces. These requirements apply to the entire CA cert chain.
  * **Evaluation Methodology:** Manufacturers MUST demonstrate compliance with the required formats and standards by providing examples of the certificate chain that is created.

### Device Keys

#### *Generation and Handling*

* **Checklist Item:**  

>   * **Requirement:** The obfuscation key ([cptra_obf_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST be generated using a method that ensures 256 bits of entropy, such as an [ESV](https://csrc.nist.rip/Projects/cryptographic-module-validation-program/entropy-validations/esv) certified entropy source or an on-die Physically Unclonable Function (PUF) that is compliant with industry standards. The CSR HMAC key ([cptra_csr_hmac_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST be generated using a method that provides full entropy across its 512-bit width and that meets the same NIST-compliant entropy standards. When either key is driven by an ECO rather than a PUF, the ECO MUST be applied prior to tapeout and the key value MUST be protected as confidential throughout its lifetime.

Reviewer 3: "the ECO MUST be applied prior to tapeout" is unnecessary and could be removed to make the actual requirement (about confidentiality) clearer.

"the key value MUST be protected as confidential throughout its lifetime" needs additional clarification, as it does not provide testable evaluation criteria. Recommend moving this to a separate requirement line item.

Would recommend moving The CSR HMAC key to its own requirement line item as well.

Author: B Keen will work on revising this language

Author: ECO verbiage addressed in https://github.com/chipsalliance/Caliptra/pull/630/commits/45fa8902ce5d725e7eca7b99133fb813f57128fe

Author: Split obf key and CSR Hmac, + added HSM example verbiage in https://github.com/chipsalliance/Caliptra/pull/630/commits/c4624d13afc7afa396edff099f61006ab8d51d71.

Author: @syncsrc-nv please mark this resolved if you're ok with the edit.

  * **Evaluation Methodology:** Manufacturers MUST demonstrate, through architectural documentation, security analyses, and verifiable testing, that the obfuscation key and CSR HMAC key are inaccessible to firmware and other unauthorized components and are not exposed on scan, debug, or trace paths. Acceptable evidence SHOULD cover each of the following categories:
    * **DFT intent.** Synthesis or DFT-tool constraint files demonstrating that the flops latching each key (for example, `cptra_scan_mode_Latched_d/f`, `field_storage.internal_obf_key`, and `cptra_csr_hmac_key_reg`) are explicitly excluded from scan insertion.
    * **DFT result.** Scan-insertion reports, scan-chain stitching reports, or gate-level netlist diffs confirming that the named cells are absent from every scan chain in the final taped-out netlist.

>     * **Dynamic verification.** Gate-level simulation with scan enable asserted showing that key state cannot be observed at any scan output, debug interface, or trace port.

Reviewer 3: I don't know that this needs to change since it isn't mandatory, but proving negatives is rarely feasible.

Author: The language is not written as requiring "proof". The acceptable evidence is already written as a SHOULD, and the idea here is to require integrators to provide evidence that they have completed due diligence to verify that the requirement was followed. It's up to the SRP to accept that argument.

>     * **Silicon validation.** Results from production scan-ATPG patterns executed during first-silicon (A0) bring-up or wafer/package test, demonstrating that the key flops are not present on any observable scan chain.

Reviewer 3: Not sure what this is implying, test patterns are carefully crafted for efficiency and will not do an exhaustive search. A negative result from such tests does not provide a guarantee of non-reachability.

Author: Could you recommend some alternative language? The goal here is to give guidance on possible types of evidence that integrators can provide to an SRP to demonstrate compliance with scan-chain exclusion, not to require a proof.

* **Checklist Item:**  
  * **Requirement:** The CSR HMAC key ([cptra_csr_hmac_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST be generated using a method that provides full entropy across its 512-bit width, such as an [ESV](https://csrc.nist.rip/Projects/cryptographic-module-validation-program/entropy-validations/esv) certified entropy source. The key value MUST be protected as confidential throughout its lifetime, as it protects the integrity of the CSR by binding it to the mask set.
  * **Evaluation Methodology:** Manufacturers MUST provide documentation on the generation method for the CSR HMAC key, including entropy measurements, applicable standards compliance, and the manufacturing controls used to keep the key confidential.
    * **Example:** Establish the CSR HMAC Key in an HSM. After the ECO is completed the key should be permitted only to exist inside HSMs and external plaintext copies should be destroyed.
* **Checklist Item:**  

>   * **Requirement:** The obfuscation key ([cptra_obf_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) and CSR HMAC key ([cptra_csr_hmac_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST NOT be accessible (readable or modifiable) to firmware or any on-chip non-Caliptra entities, including preventing oracle attacks. Both keys MUST NOT be on any scannable path, and the key signals themselves MUST NOT be observable outside of Caliptra through debug, trace, scandump, or any other on-die or off-die mechanism.

Reviewer: Maybe this would be easier to word and easier to verify as two separate requirements. One for the OBF key and one for the HMAC key. WDYT?

Author: Please leave a `suggestion` with the proposed change.

Reviewer: Never mind, I couldn't find a very clean way to do it. I'll drop this comment.

  * **Requirement:** The IDEVID certificates, including CA certificates, MUST adhere to specified formats and cryptographic standards (e.g. key sizes and algorithms as per NIST guidelines). Acceptable algorithms and key strengths MUST meet or exceed the security strength that the Caliptra IP produces. These requirements apply to the entire CA cert chain.
  * **Evaluation Methodology:** Manufacturers MUST demonstrate compliance with the required formats and standards by providing examples of the certificate chain that is created.

### *Device Keys*

#### *Generation and Handling*

* **Checklist Item:**  

>   * **Requirement:** The obfuscation key ([cptra_obf_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST be generated using a method that ensures 256 bits of entropy, such as an [ESV](https://csrc.nist.rip/Projects/cryptographic-module-validation-program/entropy-validations/esv) certified entropy source or an on-die Physically Unclonable Function (PUF) that is compliant with industry standards. When driven by an ECO rather than a PUF the key value MUST be protected as confidential throughout its lifetime and destroyed after it is no longer needed.

Reviewer 3: - changes to the key confidentiality wording to mirror the wording for key protection used in the FW Authentication Keys section.
- additional specificity on requirement around key destruction and retention
- split out into multiple line-items
```suggestion
  * **Requirement:** The obfuscation key ([cptra_obf_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST be generated using a method that ensures 256 bits of entropy, such as an [ESV](https://csrc.nist.rip/Projects/cryptographic-module-validation-program/entropy-validations/esv) certified entropy source or an on-die Physically Unclonable Function (PUF) that is compliant with industry standards. 
  * **Requirement:** When driven by an ECO rather than a PUF access to the key value MUST be restricted to only authorized entities, and MUST be stored in a manner that prevents disclosure. 
  * **Requirement:** Design, test, manufacturing, and other processes that require the unencrypted key value MUST NOT store or retain it past completion.
```

  * **Evaluation Methodology:** Manufacturers MUST provide documentation on the generation method for the obfuscation key, including entropy measurements, applicable standards compliance, and, where the key is delivered via ECO, the manufacturing controls used to keep the key confidential.
* **Checklist Item:**  

>   * **Requirement:** The CSR HMAC key ([cptra_csr_hmac_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST be generated using a method that provides full entropy across its 512-bit width, such as an [ESV](https://csrc.nist.rip/Projects/cryptographic-module-validation-program/entropy-validations/esv) certified entropy source. The key value MUST be protected as confidential throughout its lifetime, as it protects the integrity of the CSR by binding it to the mask set.

Reviewer 3: same change as suggested for the obfuscation key.
```suggestion
  * **Requirement:** The CSR HMAC key ([cptra_csr_hmac_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST be generated using a method that provides full entropy across its 512-bit width, such as an [ESV](https://csrc.nist.rip/Projects/cryptographic-module-validation-program/entropy-validations/esv) certified entropy source.
  * **Requirement:** Access to the key value MUST be restricted to only authorized entities, and MUST be stored in a manner that prevents disclosure.
```

Author: I edited your suggestion before applying to correct the indentation on the second bullet.

  * **Evaluation Methodology:** Manufacturers MUST provide documentation on the generation method for the CSR HMAC key, including entropy measurements, applicable standards compliance, and the manufacturing controls used to keep the key confidential.

>     * **Example:** Establish the CSR HMAC Key in an HSM. After the ECO is completed the key should be permitted only to exist inside HSMs and external plaintext copies should be destroyed.

Reviewer 3: "only exist inside an HSM" could be interpreted as restricting even necessary validation/manufacturing flows.  
```suggestion
    * **Example:** Establish the CSR HMAC Key in an HSM. After the ECO is completed plaintext copies should be destroyed.
```

* **Checklist Item:**  
  * **Requirement:** The obfuscation key ([cptra_obf_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) and CSR HMAC key ([cptra_csr_hmac_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST NOT be accessible (readable or modifiable) to firmware or any on-chip non-Caliptra entities, including preventing oracle attacks. Both keys MUST NOT be on any scannable path, and the key signals themselves MUST NOT be observable outside of Caliptra through debug, trace, scandump, or any other on-die or off-die mechanism.

>   * **Evaluation Methodology:** Manufacturers MUST demonstrate, through architectural documentation, security analyses, and verifiable testing, that the obfuscation key and CSR HMAC key are inaccessible to firmware and other unauthorized components and are not exposed on scan, debug, or trace paths. Acceptable evidence SHOULD cover each of the following categories:

Reviewer 3: I'm uncomfortable with the implication that evidence of all the example types should be provided, given the limitations and non-definitive nature of things like dynamic and post-Si tests. 
```suggestion
  * **Evaluation Methodology:** Manufacturers MUST demonstrate, through architectural documentation, security analyses, and verifiable testing, that the obfuscation key and CSR HMAC key are inaccessible to firmware and other unauthorized components and are not exposed on scan, debug, or trace paths. Acceptable evidence SHOULD cover:
```

Author: Your edit seems reasonable

    * **DFT intent.** Synthesis or DFT-tool constraint files demonstrating that the flops latching each key (for example, `cptra_scan_mode_Latched_d/f`, `field_storage.internal_obf_key`, and `cptra_csr_hmac_key_reg`) are explicitly excluded from scan insertion.
    * **DFT result.** Scan-insertion reports, scan-chain stitching reports, or gate-level netlist diffs confirming that the named cells are absent from every scan chain in the final taped-out netlist.

>     * **Dynamic verification.** Gate-level simulation with scan enable asserted showing that key state cannot be observed at any scan output, debug interface, or trace port.

Reviewer 3: "dynamic" is a poor word to use here, as static/taint analysis would be more useful at producing this result. 
```suggestion
    * **Verification result.** Gate-level simulation with scan enable asserted showing that key state cannot be observed at any scan output, debug interface, or trace port.
```

>     * **Silicon validation.** Results from production scan-ATPG patterns executed during first-silicon (A0) bring-up or wafer/package test, demonstrating that the key flops are not present on any observable scan chain.

Reviewer 3: This example should be removed, post-Si testing cannot provide such evidence unless the correct key value was known to the tester. 
```suggestion
```

  * **Requirement:** When driven by an ECO rather than a PUF access to the key value MUST be restricted to only authorized entities, and MUST be stored in a manner that prevents disclosure. 
  * **Requirement:** Design, test, manufacturing, and other processes that require the unencrypted key value MUST NOT store or retain it past completion.
  * **Evaluation Methodology:** Manufacturers MUST provide documentation on the generation method for the obfuscation key, including entropy measurements, applicable standards compliance, and, where the key is delivered via ECO, the manufacturing controls used to keep the key confidential.
* **Checklist Item:**  
  * **Requirement:** The CSR HMAC key ([cptra_csr_hmac_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST be generated using a method that provides full entropy across its 512-bit width, such as an [ESV](https://csrc.nist.rip/Projects/cryptographic-module-validation-program/entropy-validations/esv) certified entropy source.

>   * **Requirement:** Access to the key value MUST be restricted to only authorized entities, and MUST be stored in a manner that prevents disclosure.

Reviewer 3: Add requirement for CSR-HMAC to mirror list of obfuscation key requirements. 
```suggestion
  * **Requirement:** Access to the key value MUST be restricted to only authorized entities, and MUST be stored in a manner that prevents disclosure.
  * **Requirement:** Design, test, manufacturing, and other processes that require the unencrypted key value MUST NOT store or retain it past completion except as required for certificate issuance.
```

  * **Evaluation Methodology:** Manufacturers MUST demonstrate compliance with the required formats and standards by providing examples of the certificate chain that is created.

### *Device Keys*

#### *Generation and Handling*

* **Checklist Item:**  
  * **Requirement:** The obfuscation key ([cptra_obf_key](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/CaliptraIntegrationSpecification.md#interface)) MUST be generated using a method that ensures 256 bits of entropy, such as an [ESV](https://csrc.nist.rip/Projects/cryptographic-module-validation-program/entropy-validations/esv) certified entropy source or an on-die Physically Unclonable Function (PUF) that is compliant with industry standards. 
  * **Requirement:** When driven by an ECO rather than a PUF access to the key value MUST be restricted to only authorized entities, and MUST be stored in a manner that prevents disclosure. 
  * **Requirement:** Design, test, manufacturing, and other processes that require the unencrypted key value MUST NOT store or retain it past completion.

>   * **Evaluation Methodology:** Manufacturers MUST provide documentation on the generation method for the obfuscation key, including entropy measurements, applicable standards compliance, and, where the key is delivered via ECO, the manufacturing controls used to keep the key confidential.

Reviewer 4: Please comment about delivering the obf key through encrypted RTL as an alternative method? Thanks

Author: Encrypted RTL isn't a model for defining the obf key that we've contemplated, so there is currently no comment about it. The integration requirements are PUF or ECO of the obf key.
The fundamental requirement is that the value must be kept secret from any leakage during the design process. Even with Encrypted RTL, is the key constant visible from the synthesized netlist? That would be a problem. Using ECO delays the obf key definition until the last step to avoid this.
Do you propose that you can provide the same, or stronger, security guarantees with encrypted RTL as if the key were defined through ECO or PUF?

Reviewer 4: No. Just checking to see if it is a possibility. Even with ECO, there is still a netlist containing the final values. Both methods seem to be similar in term of security strength.

Author: Yes, the intent of the ECO is to limit the duration of time where the keys are present and reduce the number of generated reports/tooling exposure of the keys. It still results in netlists with the keys, but to the minimal extent necessary. With an encrypted RTL key, I suppose you'd add it to the design much earlier and the entire design cycle would carry that exposure. So for now it's not something we want to allow.

Author: I am marking this discussion resolved. Please contact your chip customer (e.g. Microsoft) directly if you have follow up questions on integration details that are not explicitly called out in these requirements. E.g., while we persist the requirement for ECO or PUF on the obf key, your customer will be interested to hear about the various measures you've taken to protect secrets during manufacturing. That's often the appropriate channel for these types of discussions where there's ambiguity in the Trademark requirements.

### *Hardware Implementation*

* **Checklist Item:**  

>   * **Requirement:** Hardware implementation of Caliptra MUST adhere to the SoC Integration Requirements of the RTL release that was used. Refer to docs/CaliptraIntegrationSpecification.md (section "SoC integration requirements") from the official release that was consumed for integration, as listed in [Releases](https://github.com/chipsalliance/caliptra-rtl/releases). Integrators pursuing the **Caliptra Subsystem Trademark** must additionally consult the Caliptra Subsystem Integration Specification for Subsystem-specific integration requirements.

Reviewer: nit: should we add a link to the subsystem releases as well?

Author: addressed

  * **Requirement:** SoC firmware that interacts with Caliptra as the privileged PA_USER (for 1.X implementations) or AXI_USER (for 2.X implementations) MUST be measured, and those measurements MUST be submitted to Caliptra. Other SoC firmware SHOULD be measured. Configuration data that modifies the security properties of firmware MUST also be measured. 
  * **Requirement:** Measurements of firmware and configuration MUST be submitted to Caliptra before execution of the firmware, or usage of the configuration data. Measurements MUST be submitted to Caliptra by the same entity that collected the measurement (e.g. SoC FMC cannot pass measurements to SoC FW for submission to the Caliptra mailbox).   
  * **Evaluation Methodology:** Manufacturers MUST provide a detailed description of how measurements are communicated to Caliptra.  
* **Checklist Item:**

>   * **Requirement:** Caliptra MUST be a non-bypassable element of the SoC Trusted Computing Base. Successful completion of Caliptra's boot MUST be a necessary precondition for the SoC to reach an operational state on every reset and boot path, and the SoC MUST NOT provide any mechanism to bypass, disable, hold in reset, or ignore the results of Caliptra in production silicon. Mechanisms whose sole effect is to enable Caliptra are permitted only if they are irreversibly committed to the enabled state and hardened against side-channel attacks by the manufacturer prior to shipment.

Reviewer 3: This cannot be a MUST. The same tapeout could be used to ship Caliptra-disabled SKUs which do not receive the Trademark.

Author: This multi SKU use case is not precluded by the doc language. The statement 
> Mechanisms whose sole effect is to enable Caliptra are permitted only if they are irreversibly committed to the enabled state and hardened against side-channel attacks by the manufacturer prior to shipment.

permits vendors to add mechanisms to conditionally bypass Caliptra, as long as it is done during manufacturing and is irrevocable once deployed. Ability to bypass Caliptra in deployed silicon would be a major problem.

Author: Ultimately needs to be a discussion with Product owner/SKU generation. Not a Trademark enforceable requirement.

Author: Moved to a subsystem-specific requirement (for Caliptra Core, product vendors need to discuss implementation around this with the end customer). For Subsystem, which operates in active mode, it's a strict requirement that Caliptra be engaged in the boot process.

Reviewer 5: @calebofearth would it be possible to have informative language that addresses a previous comment you made around the permissibility of vendors to configure Caliptra in a Manufacturing environment? The point being we would like it to be clear to the Compliance evaluator that supporting such a capability should not disqualify the vendor from achieving Trademark.  I'm suggesting the informative language be an addition to what you've currently specified (not suggesting a change to this wording).

Author: @gamilc Is this informative note not sufficient?
>Mechanisms whose sole effect is to enable Caliptra are permitted only if they are irreversibly committed to the enabled state and hardened against side-channel attacks by the **_manufacturer prior to shipment._**

Or do you just request an informative note in the Caliptra Core requirements section?
This checklist is about defining requirements, not giving advice or informative notes. If there is no requirement around this in the Caliptra Core section, inference for SRP to make is that it's not a trademark requirement for Caliptra Core (passive mode) integrations.

---

# Part II: Caliptra Core Trademark — Additional Requirements (Passive Mode)

The requirements in this section apply **only** to integrations pursuing the
**Caliptra Core Trademark (Passive Mode)**. All requirements in Part I also
apply.

## RTL Configuration

### *Core Mode*

* **Checklist Item:**
  * **Requirement:** The macro `CALIPTRA_MODE_SUBSYSTEM` MUST NOT be defined in
    `config_defines.svh` or in any build process. Defining this macro activates
    Subsystem-mode behavior and is incompatible with a Caliptra Core Trademark integration.
  * **Evaluation Methodology:** Manufacturers MUST provide the `config_defines.svh`
    used in the integration and demonstrate that `CALIPTRA_MODE_SUBSYSTEM` is
    absent.

### *TRNG Configuration (Core)*

* **Checklist Item:**
  * **Requirement:** The internal TRNG (`CALIPTRA_INTERNAL_TRNG`) is optional in
    Core Trademark integrations. If the internal TRNG is not enabled, the SoC
    MUST provide a compliant external TRNG via the Caliptra TRNG HW API. If the
    internal TRNG is used, the TRNG self-test threshold registers

>     (`CPTRA_iTRNG_ENTROPY_CONFIG0` and `CPTRA_iTRNG_ENTROPY_CONFIG1`) MUST NOT

Reviewer: fyi: Caliptra 2.0.1+ ROM can take a `0` value for these registers. In https://github.com/chipsalliance/caliptra-sw/blob/main/drivers/src/csrng.rs#L470 it will set the threshold to a default value when the registers are set to `0`

Author: fixed in https://github.com/chipsalliance/Caliptra/pull/630/commits/45fa8902ce5d725e7eca7b99133fb813f57128fe

    tied to ground.

### *SHA Accelerator Access Restriction*

* **Checklist Item:**
  * **Requirement:** The SHA accelerator block MUST be inaccessible to SoC
    components when Caliptra Core is integrated in Passive mode, if integrating
    version 2.0 or later. In such designs, AXI streaming mode is disabled; the SHA
    accelerator is available only in Caliptra-internal mailbox mode. No SoC
    agent may access the SHA accelerator via any path.
  * **Evaluation Methodology:** Manufacturers MUST demonstrate through RTL
    review or architecture documentation that no SoC agent has a path to the
    SHA accelerator, and that AXI streaming mode is not enabled. The value
    driven for the strap `strap_ss_caliptra_dma_axi_user` is used as a
    gating condition for SHA accelerator access via AXI. There shall
    not be a path from any SoC AXI manager that drives this AXI_USER value
    to Caliptra Core's AXI subordinate interface. This requirement applies
    for Caliptra Core release versions 2.0.0 and later.

### *Subsystem Strap Isolation*

* **Checklist Item:**
  * **Requirement:** All `strap_ss_*` Subsystem-mode strap signals (except for

>     strap_ss_caliptra_dma_axi_user) MUST be tied to logic 0.

Reviewer: nit: The rest of these signals are written with codeblocks
```suggestion
    `strap_ss_caliptra_dma_axi_user`) MUST be tied to logic 0.
```

### *SHA Accelerator Access Restriction (Subsystem)*

* **Checklist Item:**
  * **Requirement:** The SHA accelerator MUST be accessible only via Caliptra’s
    AXI DMA block. No SoC agent may access the SHA accelerator directly. The
    SHA accelerator enforces this restriction via AXI AxUSER checking; the
    integrator MUST configure `strap_ss_caliptra_dma_axi_user` to correctly
    identify Caliptra’s DMA transactions.
  * **Evaluation Methodology:** Manufacturers MUST provide strap configuration
    documentation and demonstrate, via RTL or architecture review, that no
    direct SoC path to the SHA accelerator exists and that AxUSER checking is
    correctly configured.

### *Subsystem Straps Configuration*

* **Checklist Item:**
  * **Requirement:** All `strap_ss_*` Subsystem-mode strap signals MUST be
    properly configured per the Caliptra Subsystem Integration Specification.
  * **Evaluation Methodology:** Manufacturers MUST provide strap configuration
    documentation showing specification-compliant values for all
    `strap_ss_*` signals.

## OCP Streaming Boot

* **Checklist Item:**
  * **Requirement:** OCP streaming boot via the I3C interface MUST be implemented

>     in compliance with the OCP Recovery specification.

Reviewer 3: Are the I3C interface and recovery flows provided by the subsystem RTL/ROM? what implementation activities does the integrator need to do in order to meet this requirement?

Author: Will add some clarification around I3C  vs AXI streaming boot methodologies and integrator requirements

Author: Rewording this, but basically the requirement is that integrators must utilize the subsystem-provided I3C core and recovery interface logic rather than trying to implement it on their own outside of the Caliptra Subsystem boundary.

* **Checklist Item:**  
  * **Requirement:** There MUST be a defined process for handling reported vulnerabilities, including assessment, communication, and deployment of security updates or mitigations.  
  * **Evaluation Methodology:** Manufacturers MUST provide their flaw remediation policy. This policy must describe how vulnerabilities are received (e.g., bug bounty programs), evaluated, and prioritized, and how updates are securely delivered to devices in the field. Procedures for notifying stakeholders and compliance with relevant regulations should also be described.

---

> # Part II: Caliptra Core Trademark — Additional Requirements (Passive Mode)

Author: Passive is used here for clarification. Core Mode implies that Caliptra is passive. Do you know if we defined this in any official document?

> # Part II: Caliptra Core Trademark — Additional Requirements (Passive Mode)

Reviewer: I think I was getting my terms mixed up and picturing the "Legacy" terminology listed below. Closing this comment

## Mailbox and External Staging Area

* **Checklist Item:**
  * **Requirement:** The Caliptra Core mailbox SRAM MUST be configured to 16 KiB.
    An external staging area MUST be provisioned and accessible to Caliptra via
    the AXI DMA interface for firmware image transfer.
  * **Evaluation Methodology:** Manufacturers MUST provide memory map
    documentation confirming the 16 KiB mailbox size and demonstrating that an
    appropriately sized and accessible external staging area is configured.

## Subsystem Straps Configuration

* **Checklist Item:**
  * **Requirement:** All `strap_ss_*` Subsystem-mode strap signals MUST be
    properly configured per the Caliptra Subsystem Integration Specification.
    These signals MUST NOT be tied to 0.
  * **Evaluation Methodology:** Manufacturers MUST provide strap configuration
    documentation showing non-zero, specification-compliant values for all
    `strap_ss_*` signals.

## OCP Streaming Boot

* **Checklist Item:**
  * **Requirement:** OCP streaming boot via the I3C interface MUST be implemented
    in compliance with the OCP Recovery specification.
  * **Evaluation Methodology:** Manufacturers MUST provide documentation or test

>     evidence demonstrating compliance with the OCP Recovery specification for

Reviewer 3: Does OCP provide a test suite or procedure for this that can be linked?

Author: I'm unsure. Perhaps all we need to require here is that the integration follows our spec.

Author: @syncsrc-nv can we resolve this comment? In its current manifestation the language just says integrators must use our I3C core and implement streaming boot through our defined paths. That tighter constraint should address your concerns about demonstrating compliance.

Reviewer 3: Yeah, agree this is for OCP to solve, not Caliptra.

    * **Dynamic verification.** Gate-level simulation with scan enable asserted showing that the identified signals cannot be observed at any scan output, debug interface, or trace port.
    * **Silicon validation.** Results from production scan-ATPG patterns executed during first-silicon (A0) bring-up or wafer/package test, demonstrating that the identified flops are not present on any observable scan chain.

## Subsystem Fuse Map (TODO update this verbiage)

* **Checklist Item:**
  * **Requirement:** The integrator MUST implement the Caliptra Subsystem fuse
    map including all fuses defined in the Caliptra Core fuse map and (... FIXME ...).
    Refer to the fuse map section of the Caliptra Subsystem
    Integration Specification.
  * **Evaluation Methodology:** Manufacturers MUST provide fuse map
    documentation demonstrating that both the Caliptra Core fuse map and the
    Subsystem-specific fuse map are fully implemented.

## Secure Processes

### Development Process

#### *Integrity of Hardware and ROM*

* **Checklist Item:**

>   * **Requirement:** The full Caliptra Subsystem RTL MUST be from an official release

Reviewer 6: When the obfuscated UDS seed comes from a PUF (instantiated outside the subsystem boundary) instead of fuses, subsystem RTL will need to be modified as the UDS no longer comes from within the subsystem OTP controller.  In this case, would the code still need to be published to GitHub for TAC approval to meet this requirement?

Author: Yes, any changes to RTL must go through the exception request and must receive approval. Process is being defined in this PR: https://github.com/chipsalliance/Caliptra/pull/692/changes

Reviewer 3: Note: it should be possible to use PUF for the de-obfuscation key without RTL change. If integrators are interested in using PUF in other ways, I would recommend they file an RFC for the RTL changes to support those alternatives.

    [Releases](https://github.com/chipsalliance/caliptra-ss/releases).
  * **Evaluation Methodology:** Manufacturers MUST show all changes made to Subsystem
    RTL files. Manufacturers MUST show that changes to files in the Integrator RTL
    modification list are limited to the scope described in those requirements, and
    MUST show TAC approval for other changes.
* **Checklist Item:**
  * **Requirement:** Cryptographic hash of the Caliptra Core ROM used in final hardware design MUST match reference hash of official Caliptra Core releases.
  * **Evaluation Methodology:** Manufacturers MUST demonstrate that the SHA384 hash of ROM matches that of the official release that is integrated.

---

# Part IV: Caliptra Subsystem with OCP L.O.C.K. Trademark - Additional Requirements

The requirements in this section apply **only** to integrations pursuing the
Trademark for Caliptra Subsystem in the OCP L.O.C.K.-enabled configuration.
All requirements in [Part I](#part-i-common-requirements) and [Part III](#part-iii-caliptra-subsystem-trademark---additional-requirements) also apply.

## OCP L.O.C.K. Specification Compliance

* **Checklist Item:**

>   * **Requirement:** Any product integrating Caliptra Subsystem configured with support for OCP L.O.C.K. MUST comply with all requirements listed in [lock_spec.ocp](../ocp_lock/lock_spec.ocp).

Reviewer 3: This needs to make some clear statement about how to identify what in that doc are requirements that need to be part of the audit, such as referencing the Compliance section.

Wording here also needs to account for versioning of the LOCK requirements. Is LOCK versioning captured in the release list referenced by other requirements?

Author: Matt, could you use the "suggestion" feature to recommend the specific verbiage you'd like to see here?

Author: Accepted your suggestion. Please mark this resolved if it looks good to you.

>   * **Requirement:** Any product integrating Caliptra Subsystem configured with support for OCP L.O.C.K. MUST comply with all requirements listed in [lock_spec.ocp](../ocp_lock/lock_spec.ocp).

Reviewer 7: Should this point to a specific revision, instead of whatever is the latest draft?

Author: Please use a `suggestion` to provide specific text updates you'd like to see.

Author: See https://github.com/chipsalliance/Caliptra/pull/630/changes#r3482983596

>   * **Requirement:** Any product integrating Caliptra Subsystem configured with support for OCP L.O.C.K. MUST comply with all requirements listed in [lock_spec.ocp](../ocp_lock/lock_spec.ocp).

Reviewer 3: ```suggestion
  * **Requirement:** Any product integrating Caliptra Subsystem configured with support for OCP L.O.C.K. MUST comply with the requirements listed in the Compliance section of [lock_spec.ocp](../ocp_lock/lock_spec.ocp).
```

>   * **Requirement:** Any product integrating Caliptra Subsystem configured with support for OCP L.O.C.K. MUST comply with the requirements listed in the Compliance section of [lock_spec.ocp](../ocp_lock/lock_spec.ocp).

Author: ```suggestion
  * **Requirement:** Any product integrating Caliptra Subsystem configured with support for OCP L.O.C.K. MUST comply with the requirements listed in Section 4.10 Compliance of [OCP L.O.C.K. Specification v1.1](https://www.opencompute.org/documents/ocp-lock-specification-v1-1-final-pdf).
```

Author: @clundin25 @syncsrc-nv FYI

Reviewer 7: LGTM, thank you!

Reviewer 3: looks good to me

* **Checklist Item:**
  * **Requirement:** The integrator MUST implement the full Caliptra Subsystem fuse map per
    the [Fuse Macro Memory Map and Fuse Controller CSR Address Map](https://github.com/chipsalliance/caliptra-ss/releases)
    section of the Caliptra Subsystem Integration Specification from the consumed official release.
    The fuse map has three segments: **Caliptra-Core** (`CALIPTRA_CORE` prefix) fuses, which are
    mandatory and MUST conform to the
    [Caliptra Core Fuse Map](https://github.com/chipsalliance/Caliptra/blob/main/doc/Caliptra.md#fuse-map)
    with Partitions 0-5 fixed in size; **Caliptra-Subsystem** (`CALIPTRA_SS` prefix) fuses,
    which are mandatory for Subsystem integrations; and **SoC/Vendor-Specific** fuses, which
    MAY be customized or omitted.
  * **Evaluation Methodology:** Manufacturers MUST provide fuse map documentation demonstrating
    that the Caliptra-Core and Caliptra-Subsystem segments are fully implemented per the
    specifications above, and that any SoC/Vendor-Specific customizations do not affect the
    mandatory segments.

## SoC Root of Trust (RoT)

### *Boot and Initialization Process*

* **Checklist Item:**

>   * **Requirement:** Caliptra MUST be a non-bypassable element of the SoC Trusted Computing Base. Successful completion of Caliptra's boot MUST be a necessary precondition for the SoC to reach an operational state on every reset and boot path, and the SoC MUST NOT provide any mechanism to bypass, disable, hold in reset, or ignore the results of Caliptra in production silicon. Mechanisms whose sole effect is to enable Caliptra are permitted only if they are irreversibly committed to the enabled state and hardened against side-channel attacks by the manufacturer prior to shipment.

Reviewer 3: Limit the scope of this requirement to products in the documented trademark grant list defined in PR 692 and removed redundancy. 
```suggestion
  * **Requirement:** Products (defined by Vendor, Model, and Type) receiving a trademark MUST NOT be capable of bypassing, disabling, holding in reset, or ignoring the results of Caliptra in production silicon. Mechanisms whose effect is to enable Caliptra are permitted only if they are irreversibly committed to the enabled state and hardened against side-channel attacks by the manufacturer prior to shipment.
```

  * **Evaluation Methodology:** Manufacturers MUST enumerate every Subsystem-level signal, register, or storage element that carries or buffers UDS seed, Field Entropy, or Debug Unlock token material, and MUST demonstrate non-observability through the same four evidence categories used for the Device Keys access-restriction checklist item:
    * **DFT intent.** Synthesis or DFT-tool constraint files demonstrating that the identified flops are explicitly excluded from scan insertion.
    * **DFT result.** Scan-insertion reports, scan-chain stitching reports, or gate-level netlist diffs confirming that the identified cells are absent from every scan chain in the final taped-out netlist.
    * **Dynamic verification.** Gate-level simulation with scan enable asserted showing that the identified signals cannot be observed at any scan output, debug interface, or trace port.

>     * **Silicon validation.** Results from production scan-ATPG patterns executed during first-silicon (A0) bring-up or wafer/package test, demonstrating that the identified flops are not present on any observable scan chain.

Reviewer 3: Change to match requirements for CSR-HMAC and obfuscation keys. 
```suggestion
  * **Evaluation Methodology:** Manufacturers MUST enumerate every Subsystem-level signal, register, or storage element that carries or buffers UDS seed, Field Entropy, or Debug Unlock token material, and MUST demonstrate through architectural documentation, security analyses, and verifiable testing, that such signals, registers, or storage elements are inaccessible to firmware and other unauthorized components and are not exposed on scan, debug, or trace paths. Evidence SHOULD cover:
    * **DFT intent.** Synthesis or DFT-tool constraint files demonstrating that the identified flops are explicitly excluded from scan insertion.
    * **DFT result.** Scan-insertion reports, scan-chain stitching reports, or gate-level netlist diffs confirming that the identified cells are absent from every scan chain in the final taped-out netlist.
    * **Verification result.** Gate-level simulation with scan enable asserted showing that the identified signals cannot be observed at any scan output, debug interface, or trace port.
```

Author: the list of flops to exclude should not be copied verbatim from the device-keys section. I'd just remove the (for example...) bit.

Reviewer 3: Edited the suggestion to fix that, and also reverted some other unintended changes introduced by cut & paste.

## Subsystem Fuse Map

* **Checklist Item:**

>   * **Requirement:** The integrator MUST implement the full Caliptra Subsystem fuse map per

Reviewer 3: ```suggestion
  * **Requirement:** The integrator MUST implement the Caliptra Subsystem fuse map per
```
Remove "full" to avoid ambiguity, as some of the fuses are called out as being optional.

    RTL files. Manufacturers MUST show that changes to files in the Integrator RTL
    modification list are limited to the scope described in those requirements, and
    MUST show TAC approval for other changes.
* **Checklist Item:**

>   * **Requirement:** Cryptographic hash of the Caliptra ROM used in final hardware design MUST match reference hash of official Caliptra releases.

Reviewer 3: Aren't integrators expected to modify subsystem ROM?

Author: This requirement is about Caliptra ROM not MCU

Author: Subsystem still includes the Caliptra ROM, so this requirement is just saying that the same requirement from Core mode applies for Subsystem (on unmodified ROM)

Reviewer 3: "Caliptra" covers both subsystem and core, this needs to be explicit about which one it applies to if not both.

Author: ```suggestion
  * **Requirement:** Cryptographic hash of the Caliptra Core ROM used in final hardware design MUST match reference hash of official Caliptra Core releases.
```
